### PR TITLE
qgis: fix empty sourceLayerId

### LINKED ISSUE #415: Layer ID: Cannot find layer id ''
We have errornous commits with empty layer id. I suspect it happens when one edits on different map theme.

The changes are not applied. How can this be resolved?

### PR DESCRIPTION
**Pull request by miili**
This is a workaround for an empty `sourceLayerId = ""` delivered by QField, applied when the faulty JSON is loaded.

Even though the bug is (probably) in QField, the bad diff should be handled by QFieldCloud.

Closes #415

### CODE REVIEW

**Comment by miili** [comment]
@suricactus since the issue persists, maybe consider this fix.

File: docker-qgis/apply_deltas.py [hunk 1]
```diff
@@ -374,6 +374,13 @@ def delta_file_file_loader(args: DeltaOptions) -> Optional[DeltaFile]:
             obj["clientPks"],
         )
 
+        # NOTE Sometimes QField does not fill the `sourceLayerId` field
+		# In recent QGIS versions, offline editing replaces the data source of the layers, so the layer ids do not change
```
Comment (suricactus) [inline docker-qgis/apply_deltas.py:378]:
```suggestion
        # In recent QGIS versions, offline editing replaces the data source of the layers, so the layer ids do not change        
```

File: docker-qgis/apply_deltas.py [hunk 2]
```diff
@@ -374,6 +374,13 @@ def delta_file_file_loader(args: DeltaOptions) -> Optional[DeltaFile]:
             obj["clientPks"],
         )
 
+        # NOTE Sometimes QField does not fill the `sourceLayerId` field
+        # In recent QGIS versions, offline editing replaces the data source of the layers, so the layer ids do not change
```
Comment (suricactus) [inline docker-qgis/apply_deltas.py:378]:
```suggestion
        # In recent QGIS versions, offline editing replaces the data source of the layers, so the layer ids do not change
```

**Comment by miili** [comment]
@suricactus ping for workflow

**Comment by miili** [comment]
Do you want me to add a `logger.warning`?

**Comment by miili** [comment]
@suricactus can you trigger the workflow please? Having this fix in production can save me a lot of trouble and time.

**Comment by suricactus** [comment]
@miili it is part of this week plan, I just have other things to do too. ;)

**Comment by suricactus** [comment]
Added some tests.

BTW, sorry I hijacked your branch, seems GitHub allows me to edit it directly.

**Comment by suricactus** [comment]
@miili I am pretty sure you will ask when it will be deployed. Unfortunately you have to wait a bit more. But it is coming with the next version for sure!

**Comment by miili** [comment]
Thanks 🙏🏻

**Pull request merged**